### PR TITLE
Fix fetching registry index branch on nightly

### PR DESCRIPTION
**Pull request by avrong**
Uses `origin/HEAD` instead of `origin/master` to fetch the branch of registry index.

changelog: Fix loading crates local index on nightly rust

### CODE REVIEW

File: toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt [hunk 1]
```diff
@@ -64,8 +64,8 @@ class CratesLocalIndexServiceImpl
         .build()
 
     private val registryHeadCommitHash: String
-        get() = repository.resolve("origin/master")?.name ?: run {
-            LOG.error("Git revision `origin/master` cannot be resolved to any object id")
+        get() = repository.resolve("origin/HEAD")?.name ?: run {
```
Comment (Undin) [inline toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt:67]: Let's extract `origin/HEAD` into constant

**Review by Undin — APPROVED** [review]
Works for me!
Thank you for the quick investigation and fix 👍

**Review by Undin — COMMENTED** [review]
Looks like tests fail for rust 1.41

**Comment by avrong** [comment]
Cargo in toolchains uses such tags (remotes) for the registry index:
- 1.49 and older — `origin/master`;
- 1.50–1.52 — `origin/master` and `origin/HEAD`;
- 1.53 and future — `origin/HEAD`.

There are two possible solutions I see:
1. Use the toolchain to determine the Rust version and then, decide what tag should be used. But for that, we need to somehow get the project instance. I don't know if it is possible in an application-level service.
  As a solution, the service and state can be separated into project service and application service. It will also be a solution to mapping crates local index update to the cargo project updates.
2. A simpler solution is to just check both tags: if `origin/HEAD` is not found, then try `origin/master`.

**Comment by Undin** [comment]
I like the second option: it's very simple and quite efficient. Also, we usually prefer not to use concrete versions (if it's possible, of course) to support intermediate states like nightly builds

**Review by Undin — APPROVED** [review]
bors r+

**Pull request merged**